packages/react-instantsearch-dom/src/core/__tests__/createInstantSearchServer.js

```diff
@@ -920,6 +953,40 @@ describe('findResultsState', () => {
       });
     });
 
+    it('should search twice for a widget requesting initial results', async () => {
+      const getSearchParameters = jest.fn();
+      const getSearchParameters2 = jest.fn();
+
+      const Connected = createWidget({
+        getSearchParameters,
+      });
+      const Dynamic = createWidget({
+        getSearchParameters: getSearchParameters2,
+        requiresInitialResults: true,
+      });
+
+      const App = props => (
+        <InstantSearch {...props}>
+          <Dynamic />
```

Comment:
Why do we keep a `Dynamic` component in the root index in this test?

Reply:
to check it doesn't do three requests

Reply:
Not sure I understand why it would do three requests and why having it at the root prevents it.

Reply:
I'm testing that dynamic widgets does 1 extra request, and that there's no mismatch and trying to do one extra request per dynamic widget